src/main/java/mate/academy/rickandmorty/service/RickAndMortyClient.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.rickandmorty.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.external.ResponseDataDto;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class RickAndMortyClient {
+    private static final String BASE_URL = "https://rickandmortyapi.com/api/character/?page=%s";
+    private final ObjectMapper objectMapper;
+
+    public ResponseDataDto getPage(int pageNumber) {
+        HttpClient httpClient = HttpClient.newHttpClient();
```

Comment:
We are using spring, so use the IoC.
Make HttpClient the bean and inject it.

```suggestion
        HttpClient httpClient = HttpClient.newHttpClient();
```